oceannavigator/frontend/src/components/ComboBox.jsx

```diff
@@ -1,347 +1,224 @@
-import React from "react";
+import React, { useState, useEffect, useRef } from "react";
 import axios from "axios";
 import { Modal, Button, Form } from "react-bootstrap";
 import Icon from "./lib/Icon.jsx";
 import PropTypes from "prop-types";
-
-import { withTranslation } from "react-i18next";
-
-class ComboBox extends React.Component {
-  constructor(props) {
-    super(props);
-
-    // Track if mounted to prevent no-op errors with the Ajax callbacks.
-    this._mounted = false;
-
-    this.state = {
-      data: [],
-      url: null,
+import { useTranslation } from "react-i18next";
+
+function ComboBox({
+  id,
+  title,
+  url,
+  data: incomingData,
+  state: propState,
+  onUpdate,
+  multiple = false,
+  alwaysShow = false,
+  def = "",
+  children,
+}) {
+  const { t: _ } = useTranslation();
+  const isMounted = useRef(false);
+  const [optionsData, setOptionsData] = useState([]);
+  const [showHelp, setShowHelp] = useState(false);
+  const [lastUrl, setLastUrl] = useState(null);
+
+  // track mount/unmount
+  useEffect(() => {
+    isMounted.current = true;
+    return () => {
+      isMounted.current = false;
     };
+  }, []);
 
-    // Function bindings
-    this.handleChange = this.handleChange.bind(this);
-    this.showHelp = this.showHelp.bind(this);
-    this.closeHelp = this.closeHelp.bind(this);
-  }
-
-  componentDidMount() {
-    this._mounted = true; //Component mounted
-    this.populate(this.props);
-  }
-
-  componentWillUnmount() {
-    this._mounted = false; //Component not mounted
-  }
+  // re‑populate whenever url or incomingData change
+  useEffect(() => {
+    if (url !== lastUrl || incomingData !== undefined) {
+      setLastUrl(url);
+      populate();
+    }
+  }, [url, incomingData]);
```

Comment:
Probably don't need `useEffect` here since any changes to data or url will force a re-render anyway (see https://react.dev/learn/you-might-not-need-an-effect - definitely worth a read). And since these ops are pretty light and it should only re-render when state/props are changed you won't need to compare to previous values. Not sure if the intent was to recreate the functionality of `componentWillUnmount` was deprecated because it led many bad design patterns and should have been removed from our code long ago.